### DISCUSSION TITLE
MessageFormat, ResponseFormat and BodyStyle

### DISCUSSION THREAD

**Discussion by gunnistefans** [CoreWCF/CoreWCF#1179] [General]
Hello

I have a old WCF service that looks something like this,   How can I configure the RequestFormat, and the Response format in CoreWCF.

Any ideas ?

```c#
        [WebInvoke(Method = "POST", UriTemplate = "/Change", RequestFormat = WebMessageFormat.Json, ResponseFormat = WebMessageFormat.Xml, BodyStyle = WebMessageBodyStyle.Bare)]
        public void Change(Stream input)
        {
            try
            {
                var stream = new MemoryStream();
                input.CopyTo(stream);
                var writer = new StreamWriter(stream);
                writer.Flush();
                stream.Position = 0;

                string jsonContents = Encoding.UTF8.GetString(stream.ToArray());
                dynamic message = Newtonsoft.Json.JsonConvert.DeserializeObject(jsonContents);

				// Do stuff
                eventLog.WriteEntry(message.ToString(), EventLogEntryType.Information);

                stream.Position = 0;

                WebOperationContext ctx = WebOperationContext.Current;
                ctx.OutgoingResponse.StatusCode = HttpStatusCode.OK;
            }
            catch(Exception x)
            {
                WebOperationContext.Current.OutgoingResponse.ContentType = "text/plain";
                var stream = new MemoryStream();
                var writer = new StreamWriter(stream);
                writer.WriteLine(x.Message);
                writer.WriteLine(x.StackTrace);
                writer.Flush();
                stream.Position = 0;

                return;
            }
        }
```

**Comment by mconnew** [comment]
Use the CoreWCF.WebHttp package and change your using statements to CoreWCF instead of System.ServiceModel and it should work the same. Take a look at the [blog post here](https://corewcf.github.io/blog/2022/04/13/webhttp) for me details about setting the service up.

**Reply by gunnistefans** [reply]
Thanks.
That worked for me.   

But now when I run the Service locally everything works fine, on the IISExpress.    
But when I publish it to a IIS server the functions wont work.

the iis displays the /Api/ page but not /Api/Test 

can you see the problem ? 


![testfunction](https://github.com/CoreWCF/CoreWCF/assets/4387389/9bba7508-29ad-48bf-984d-845c9d5e5aa6)
![program](https://github.com/CoreWCF/CoreWCF/assets/4387389/cec63c78-54fb-4518-bdbd-33b24886c5e7)

**Comment by gunnistefans** [comment]
when opening http://URL/Api   it works .   But when I open the Http://url/Api/Test   

then it looks like the function does not exist.

**Reply by g7ed6e** [reply]
Hi @gunnistefans can you confirm that this scenario is supported with .NET Framework WCF and provide a stack trace ?

**Reply by gunnistefans** [reply]
Hello.

I never reach the state of getting an error or stack trace.    On serverside the endpoint "/Api/Test"   seams like it does not exist.   But when i run the same Localy on Local IIS it comes up.

I can send you the full project if you want.   @g7ed6e
